go/libraries/doltcore/branch_control/access.go

```diff
@@ -95,6 +107,64 @@ func (tbl *Access) GetIndex(branchExpr string, userExpr string, hostExpr string)
 	return -1
 }
 
+// Serialize writes the table to the given buffer. All encoded integers are big-endian.
+func (tbl *Access) Serialize(buffer *bytes.Buffer) {
+	tbl.RWMutex.RLock()
+	defer tbl.RWMutex.RUnlock()
+
+	// Write the version bytes
+	writeUint16(buffer, currentAccessVersion)
+	// Write the number of entries
+	numOfEntries := uint32(len(tbl.Values))
+	writeUint32(buffer, numOfEntries)
+
+	// Write the rows
+	for _, matchExpr := range tbl.Branches {
+		matchExpr.Serialize(buffer)
+	}
+	for _, matchExpr := range tbl.Users {
+		matchExpr.Serialize(buffer)
+	}
+	for _, matchExpr := range tbl.Hosts {
+		matchExpr.Serialize(buffer)
+	}
+	for _, val := range tbl.Values {
+		val.Serialize(buffer)
+	}
+	// Write the binlog
+	_ = tbl.binlog.Serialize(buffer)
+}
+
+// Deserialize populates the table with the given data. Returns an error if the data cannot be deserialized, or if the
+// table has already been written to. Deserialize must be called on an empty table.
+func (tbl *Access) Deserialize(data []byte, position *uint64) error {
```

Comment:
Typically this kind of function returns `(bytes read, error)` rather than mutating an offset pointer